### ISSUE TITLE
How to run custom function on redraw?

### ISSUE THREAD

**Issue by Milamber33** [volosoft/jtable#2201]
Hi,

I'm using a function to customise the display of some elements in my tables, and I'm running into a problem. I need to run a function to resize them appropriately any time the cell dimensions change, but I can't seem to find any event or other way to tell when this happens. All of the documented events seem to fire on DATA change, but I need to fire a function on DISPLAY change. Is there an undocumented Redraw event or anything like that? Or any other way to deal with this? What do you use to capture column resize internally?

**Comment by misterparsons** [comment]
There are no documented or undocumented events column changes. But there are ways to work around the problems. I have written a comprehensive tutorial on how to make a jtable scrollable, and this exposes the issues of column visibility changes and column width dragging. You may find it gives a clue or two.[Tutorial](http://swaleinternet.co.uk/tutorials/jTable-demo-table-scroll.php#subResize)

**Comment by Milamber33** [comment]
Thanks, I initially worked out something similar to your resize handler, but once I worked out it wasn't dealing with changes in column visibility I actually found a more elegant solution to handle both cases in the one code block. What I'm doing is extending the JQuery .css() function to trigger an event and then binding my code to that event. I got the code from here: https://stackoverflow.com/questions/2157963/is-it-possible-to-listen-to-a-style-change-event
and my implementation looks like this:
//this code creates the 'style' event type on calls to JQuery .css, have it at the top of your javascript,
//outside all other code blocks.
(function() {
    var ev = new $.Event('style'),
        orig = $.fn.css;
    $.fn.css = function() {
        var ret = orig.apply(this, arguments);
        $(this).trigger(ev);
        return ret;
    }
})();
//and to use it just put this any time after jtable load.
$("#tablename").find('th.jtable-column-header').on('style',function(){
    //your code here
});
//you should also trigger on window resize
$(window).resize(function(){
    //your code here
});

Obviously in a real implementation you'd just make a function with your code and call it on both the style event and window resize. Only catch I found is that because you're only adding the handler after load, initial setup still needs to be done separately, but that's the case with your method too. This method can also be used for a whole range of other purposes, but just be aware that it only listens for CSS changes being applied using jquery, plain javascript CSS changes will not trigger it.

Workarounds aside, it would be really good if JTable had a redraw event that could be used.

**Comment by misterparsons** [comment]
I agree column events would be the best solution, and I would implement them in a real world solution. But for a tutorial, I thought there was more to show without them.

The stackoverflow page was interesting, I was not aware of the new mutationObserver, maybe a major rewite is in order.

**Comment by Milamber33** [comment]
Yeah, I was kind of looking at that myself, but it's not supported in even IE10, and IE11 has a known issue too, so I decided it would be safer to stay away from it for now.
